wit/deps/couchbase/subdocument.wit

```diff
@@ -0,0 +1,253 @@
+    /// Remove the field at the given path
+    remove(tuple<subdocument-path, remove-options>),
+
+    /// Increment a counter value
+    increment(tuple<subdocument-path, s64, increment-options>),
+    /// Decrement a counter value
+    decrement(tuple<subdocument-path, s64, decrement-options>),
+
+    /// Perform a dictionary add unique operation
+    array-add-unique(tuple<subdocument-path, array-add-unique-options>),
+    /// Add element(s) to the end (i.e. right) of an array
+    array-append(tuple<subdocument-path, array-append-options>),
+    /// Insert element(s) into an array, at a given position which is specified as part of the path
+    /// (ex. 'path.to.array[3]')
+    array-insert(tuple<subdocument-path, document, array-insert-options>),
+    /// Add element(s) to the beginning (i.e. left) of an array
+    array-prepend(tuple<subdocument-path, array-append-options>),
+  }
+
+  /// Errors that occur during mutate batch operations
+  variant mutate-error {
+    /// A completely unexpected error
+    unexpected(string),
+  }
+
+  /// Errors that occur during mutate for an individual subdocument path
+  variant subdocument-mutate-error {
+    /// A completely unexpected error
+    unexpected(string),
+  }
+
+  /// Options for performing batches of mutation
+  record mutate-options {
+    /// Milliseconds until the mutation operation should expire
+    expires-ms: u64,
+
+    /// Milliseconds until the mutation operation should time out
+    timeout-ms: u64,
+
+    /// CAS revision of the document
+    revision: u64,
+
+    /// Durability level for the mutation
+    ///
+    /// (this should be used instead of persist-to/replicate-to)
+    durability-level: durability-level,
+
+    /// Semantics on the store
+    ///
+    /// (this should be used instead of persist-to/replicate-to)
+    store-semantics: store-semantics,
+
+    /// How and whether to retry the operation
+    retry-strategy: option<retry-strategy>,
+
+    /// A known span to associate this lookup with
+    parent-span: option<request-span>,
+
+    /// Whether to preserve expiry
+    preserve-expiry: bool,
+
+    /// (Deprecated)
+    /// Traditional durability settings for persistence, only usable from Couchbase versions earlier than 6.0
+    ///
+    /// See: https://docs.couchbase.com/go-sdk/current/howtos/subdocument-operations.html#1.6@go-sdk::durability.adoc
+    persist-to: u64,
+
+    /// (Deprecated)
+    /// Traditional durability settings for replication, only usable from Couchbase versions earlier than 6.0
+    ///
+    /// See: https://docs.couchbase.com/go-sdk/current/howtos/subdocument-operations.html#1.6@go-sdk::durability.adoc
+    replicate-to: u64,
+  }
+
+  /// Perform a mutate of one or more specs in a given document
+  mutate: func(
+    id: document-id,
+    operations: list<mutate-operation>,
+    options: mutate-options,
+  ) -> result<list<result<document, subdocument-mutate-error>>, mutate-error>;
```

Comment:
Include `Cas` in the result ?

Reply:
Absolutely -- this should be a `mutation-result` here:

- [ ] Add `mutation-result` record that maps properly to `MutationResult`